Source/Core/VideoBackends/Software/Rasterizer.cpp

```diff
@@ -83,6 +70,19 @@ void Init()
 	ZSlope.f0 = 1.f;
 }
 
+// Returns approximation of log2(f) in s28.4
+// results are close enough to use for LOD
+static s32 FixedLog2(float f)
+{
+	u32 x;
+	std::memcpy(&x, &f, sizeof(u32));
+
+	s32 logInt = ((x & 0x7F800000) >> 19) - 2032; // integer part
+	s32 logFract = (x & 0x007fffff) >> 19; // approximate fractional part
```

Comment:
Yeah, initially  I would have done that, but reading up on C++ behavior. It would be considered undefined behavior (Section 9.5 point 1 states, and I copy verbatim from the standard): 

> In a union, at most one of the non-static data members can be active at any time, that is, the value of at
most one of the non-static data members can be stored in a union at any time. [ Note: One special guarantee
is made in order to simplify the use of unions: If a standard-layout union contains several standard-layout
structs that share a common initial sequence (9.2), and if an object of this standard-layout union type
contains one of the standard-layout structs, it is permitted to inspect the common initial sequence of any of
standard-layout struct members; see 9.2. —end note ]